File: livingrimoire start here/LivinGrimoire java/src/Auxiliary_Modules/TrgEveryNMinutes.java
package Auxiliary_Modules;

public class TrgEveryNMinutes extends TrGEV3{
    // trigger returns true every minutes interval, post start time
    private int minutes; // minute interval between triggerings
    private final TrgTime trgTime;
    private String timeStamp;

    public TrgEveryNMinutes(String startTime, int minutes) {
        this.minutes = minutes;
        this.timeStamp = startTime;
        trgTime = new TrgTime();
        trgTime.setTime(startTime);
    }

    public void setMinutes(int minutes) {
            if (minutes > -1) {
            this.minutes = minutes;}
    }

    @Override
    public Boolean trigger() {
        if (trgTime.alarm()){
            timeStamp = TimeUtils.getFutureInXMin(minutes);
            trgTime.setTime(timeStamp);
            return true;
        }
        return false;
    }

    @Override
    public void reset() {
        timeStamp = TimeUtils.getCurrentTimeStamp();
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/auxiliary_modules/TrgEveryNMinutes.kt
package auxiliary_modules

class TrgEveryNMinutes(
    startTime: String, // trigger returns true every minutes interval, post start time
    private var minutes // minute interval between triggerings
    : Int
) : TrGEV3() {
    private val trgTime: TrgTime
    private var timeStamp = ""

    init {
        timeStamp = startTime
        trgTime = TrgTime()
        trgTime.setTime(startTime)
    }

    fun setMinutes(minutes: Int) {
        if (minutes > -1) {
            this.minutes = minutes
        }
    }

    override fun trigger(): Boolean {
        if (trgTime.alarm()) {
            timeStamp = TimeUtils.getFutureInXMin(minutes)
            trgTime.setTime(timeStamp)
            return true
        }
        return false
    }

    override fun reset() {
        timeStamp = TimeUtils.currentTimeStamp
    }
}